edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/WorkloadClientVersioned.cs

```diff
@@ -55,21 +59,51 @@ protected internal async Task<T> Execute<T>(Func<Task<T>> func, string operation
             try
             {
                 Events.ExecutingOperation(operation, this.WorkloadUri.ToString());
-                T result = await ExecuteWithRetry(func, r => Events.RetryingOperation(operation, this.WorkloadUri.ToString(), r), this.transientErrorDetectionStrategy)
+                T result = await ExecuteWithRetry(
+                    func,
+                    r => Events.RetryingOperation(operation, this.WorkloadUri.ToString(), r),
+                    this.transientErrorDetectionStrategy)
                     .TimeoutAfter(this.operationTimeout);
                 Events.SuccessfullyExecutedOperation(operation, this.WorkloadUri.ToString());
+                this.failedSocketCount = 0;
                 return result;
             }
+            catch (SocketException ex)
```

Comment:
catch (SocketException ex) when (ex.Message.Contains("Connection refused") so IsWorkloadSocketStale is not needed